services/bettersocial/post/createPostV3.js

```diff
@@ -183,6 +185,50 @@ function processAnonymous(isAnonimous, body, data) {
   return data;
 }
 
+async function processSendSystemMessage(
+  filteredTopics,
+  selfUserId,
+  username,
+  with_system_message = false,
+  isAnonimous = false
+) {
+  if (!with_system_message) return;
+  if (!filteredTopics) return;
+  if (filteredTopics?.length == 0) return;
+  if (!selfUserId || !username) throw new Error('Missing params');
+
+  console.log('send system message');
+
+  const adminClient = new StreamChat(
+    Environment.GETSTREAM_API_KEY,
+    Environment.GETSTREAM_API_SECRET
+  );
+
+  adminClient.connectUser({id: selfUserId});
+  const promises = filteredTopics?.map((topic) => {
+    return new Promise((resolve, reject) => {
+      const channel = adminClient.channel('topics', `topic_${topic}`);
+
+      channel
+        .create()
+        .then(() => {
+          Getstream.chat
+            .sendCreatePostTopicSystemMessage(channel, selfUserId, username, {isAnonimous})
+            .then(() => {
+              resolve();
+            })
+            .catch((e) => {
+              reject(e);
+            });
+        })
+        .catch((e) => reject(e));
+    });
+  });
+
+  await Promise.all(promises);
+  adminClient.disconnectUser();
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues in this code block:

1. The function `processSendSystemMessage` is using console.log for logging which is not recommended for production code. It's better to use a proper logging library that can be configured to change the log level dynamically and write logs to various transports.

2. Error handling could be improved. Currently, if an error occurs while sending a system message for one of the topics, it will reject the promise and cause `Promise.all(promises)` to reject immediately. This means that if there's an error with one topic, no further topics will be processed. Instead, you might want to catch errors for each topic individually and continue processing the remaining topics.

3. The function `processSendSystemMessage` does not return anything. If this function is meant to be used asynchronously, it would be beneficial to return a value indicating success or failure of the operation.

Here's how you can address these issues:

```diff
-  console.log('send system message');
+  logger.info('send system message');

   const adminClient = new StreamChat(
     Environment.GETSTREAM_API_KEY,
     Environment.GETSTREAM_API_SECRET
   );

   adminClient.connectUser({id: selfUserId});
   const promises = filteredTopics?.map((topic) => {
     return new Promise((resolve) => {
       const channel = adminClient.channel('topics', `topic_${topic}`);

       channel
         .create()
         .then(() => {
           Getstream.chat
             .sendCreatePostTopicSystemMessage(channel, selfUserId, username, {isAnonimous})
             .then(() => {
               resolve();
             })
-            .catch((e) => {
-              reject(e);
-            });
+            .catch((e) => {
+              logger.error(`Error sending system message for topic ${topic}: ${e}`);
+              resolve();
+            });
         })
-        .catch((e) => reject(e));
+        .catch((e) => {
+          logger.error(`Error creating channel for topic ${topic}: ${e}`);
+          resolve();
+        });
     });
   });

   await Promise.all(promises);
   adminClient.disconnectUser();
+  return promises.length;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->